Debug-custom-ESLint-check-tests.md

```diff
@@ -0,0 +1,101 @@
+## Table of contents
+
+* [Introduction](#introduction)
+* [Print statements](#print-statements)
+* [Inspect the AST](#inspect-the-ast)
+
+## Introduction
+
+Especially for complex lint rules, it can be hard to figure out why a test failed. This page will give you some tips to help you debug your failing tests.
+
+## Print statements
+
+If you add a `console.log()`, `console.error()`, or `console.info()` statement to your custom ESLint rule code, the output will appear along with the test status output. For example, suppose we have a lint rule that raises an error every time `.click()` or `.sendKeys()` is called on an object instead of `action.click()` or `action.sendKeys()`, which we would like people to use instead. If we add `console.log()` statement to print `DEBUGGING found lint error` whenever our lint rule raises an error, we see this output when running the tests:
+
+```text
+  e2e-action
+    valid
+      ✓ it('should test a feature', function() {
+        action.click("Element", elem);
+      });
+      ✓ it('should test a feature', function() {
+        action.sendKeys("Element", elem, "keys");
+      });
+      ✓ it('should test a feature', function() {
+        console.log(elem.click);
+      });
+      ✓ it('should test a feature', function() {
+        console.log(elem.sendKeys);
+      });
+    invalid
+DEBUGGING found lint error
+      ✓ it('should test a feature', function() {
+        elem.click();
+      });
+DEBUGGING found lint error
+      ✓ it('should test a feature', function() {
+        elem.sendKeys("keys");
+      });
+DEBUGGING found lint error
+DEBUGGING found lint error
+      ✓ it('should test a feature', function() {
+        element(by.css('.protractor-test')).click();
+      });
+      ✓ it('should test a feature', function() {
+        element(by.css('.protractor-test')).sendKeys("keys");
+      });
+```
+
+There are a few important points right away:
```

Comment:
points --> points to note

Or perhaps better: "However, there are a few issues that you may run into with this approach:" -- that helps prepare the reader better for how the following points fit into the bigger picture.